### PR TITLE
Migrate SDK to API-PHP

### PR DESCRIPTION
**Pull request by BushraAsif**

### CODE REVIEW

File: .php_cs.cache [hunk 1]
```diff
@@ -0,0 +1 @@
...g_whitespace":true,"no_trailing_whitespace_in_comment":true,"single_blank_line_at_eof":true,"single_import_per_statement":true,"single_line_after_imports":true,"switch_case_semicolon_to_colon":true,"switch_case_space":true,"encoding":true,"full_opening_tag":true,"error_suppression":{"mute_deprecation_error":false,"noise_remaining_usages":false,"noise_remaining_usages_exclude":[]},"non_printable_character":true},"hashes":{"altapay.php":68974041,"classes\\MerchantAPI.php":2611360412,"controllers\\front\\callbackfail.php":2083725536,"controllers\\front\\callbackform.php":2508225154,"controllers\\front\\callbacknotification.php":1439388557,"controllers\\front\\callbackok.php":1184855679,"controllers\\front\\callbackopen.php":853231097,"controllers\\front\\callbackredirect.php":1435841904,"controllers\\front\\deletecreditcard.php":202155877,"controllers\\front\\orderconfirmation.php":1941333140,"controllers\\front\\payment.php":825166543,"controllers\\front\\savecreditcard.php":4088146273,"controllers\\front\\showsavedcreditcards.php":115413906,"helpers.php":1709833630,"index.php":2126644304,"lib\\altapay\\altapay-php-sdk\\lib\\helpers.php":3526644319,"Models\\Terminal.php":2650438824}}
```
Comment (AJenbo) [inline .php_cs.cache:1]: Don't commit your local cache files. Probably add this go git ignore instead.

File: controllers/front/callbackok.php [hunk 2]
```diff
@@ -50,13 +48,13 @@ public function postProcess()
...
-            if ($response->wasSuccessful()) {
+            if ($response) {
                 $orderStatus = (int) Configuration::get('PS_OS_PAYMENT');
-                $paymentType = $response->getPrimaryPayment()->getAuthType();
-                $amountPaid = $response->getPrimaryPayment()->getCapturedAmount();
-                $captureStatus = Tools::getValue('require_capture');
-                $currencyPaid = Currency::getIdByIsoCode($response->getPrimaryPayment()->getCurrency());
-                $transactionID = Tools::getValue('transaction_id');
+                $paymentType = $response->type;
+                $amountPaid = $response->Transactions[0]->CapturedAmount;
```
Comment (AJenbo) [inline controllers/front/callbackok.php:54]: We should probably check if there are any transactions before accessing the array.

File: altapay.php [hunk 3]
```diff
@@ -1392,12 +1398,25 @@ private function selectOrder($params)
...
-            $paymentDetails = $api->getPaymentDetails($paymentID);
+            $api = new API\PHP\Altapay\Api\Others\Payments(getAuth());
+            $api->setTransaction($paymentID);
+            $paymentDetails = $api->call();
+
+            $reserved = 0;
+            $captured = 0;
+            $refunded = 0;
+
+            if ($paymentDetails) {
```
Comment (AJenbo) [inline altapay.php:1412]: Can this be something else than an array? If not then there is no need to check if ti contains something as `foreach` will do so.

File: altapay.php [hunk 4]
```diff
@@ -1428,15 +1450,20 @@ public function performCapture($paymentID, $params, $captureRemainedAmount = tru
...
                     true
                 );
+
+                $api->setOrderLines($orderLines);
                 if ($statusCapture) {
-                    $api->captureAmount($paymentID, $orderLines, $orderDetail->total_paid);
+                    $api->setAmount((float) $orderDetail->total_paid);
                 } else {
-                    $api->captureAmount($paymentID, $orderLines, $amountToCapture);
+                    $api->setAmount((float) $amountToCapture);
```
Comment (AJenbo) [inline altapay.php:1458]: This is already numeric why do we need to cast it?

File: .php_cs.cache (hunk 1 above)
Comment (BushraAsif) [reply .php_cs.cache:1]: I have added this in the .gitignore file but do we still need to delete this file?

Comment (AJenbo) [reply .php_cs.cache:1]: yes, else ignoring it won't have any effect.

**Review by AJenbo — APPROVED** [review]

**Pull request merged**